### ISSUE TITLE
[java] AvoidUsingShortType false-positive when overriding methods

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#964]
**Affects PMD Version:** 6.1.0

**Rule:** [AvoidUsingShortType](https://pmd.github.io/pmd-6.1.0/pmd_rules_java_performance.html#avoidusingshorttype)

**Description:**
When overriding a inherited method, which returns a `short`, we can't avoid using short type. Hence it should be ignored.

**Code Sample demonstrating the issue:**

```
public abstract class AbstractDomNodeProxy extends AbstractNode implements org.w3c.dom.Node {
    protected final org.w3c.dom.Node node;
    //...
    @Override
    public short getNodeType() {
        return node.getNodeType();
    }
    //...
}
```
See [AbstractDomNodeProxy](https://github.com/pmd/pmd/blob/0967d5027539d8528deb025d6b6d0ecece41d8f9/pmd-xml/src/main/java/net/sourceforge/pmd/lang/xml/ast/AbstractDomNodeProxy.java#L53)

**Running PMD through:** Maven

**Comment by jsotuyod** [comment]
This seems to be a duplicate of #586, isn't it?

**Comment by adangel** [comment]
Oh, yes! Wasn't aware, we had this already :smile:
